pkg/gardenadm/translator/static_pod_test.go

```diff
@@ -0,0 +1,259 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package translator_test
+
+import (
+	"context"
+	"path/filepath"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	fakeclient "sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	extensionsv1alpha1 "github.com/gardener/gardener/pkg/apis/extensions/v1alpha1"
+	. "github.com/gardener/gardener/pkg/gardenadm/translator"
+)
+
+var _ = Describe("StaticPod", func() {
+	var (
+		ctx        = context.Background()
+		fakeClient client.Client
+	)
+
+	BeforeEach(func() {
+		fakeClient = fakeclient.NewClientBuilder().Build()
+	})
+
+	Describe("#Translate", func() {
+		When("object is a Deployment", func() {
+			var deployment *appsv1.Deployment
+
+			BeforeEach(func() {
+				deployment = &appsv1.Deployment{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        "foo",
+						Namespace:   "bar",
+						Labels:      map[string]string{"will": "be-ignored"},
+						Annotations: map[string]string{"this-as": "well"},
+					},
+					Spec: appsv1.DeploymentSpec{
+						Template: corev1.PodTemplateSpec{
+							ObjectMeta: metav1.ObjectMeta{
+								Labels:      map[string]string{"baz": "foo"},
+								Annotations: map[string]string{"bar": "baz"},
+							},
+						},
+					},
+				}
+			})
+
+			It("should successfully translate a pod w/o volumes", func() {
+				files, err := Translate(ctx, fakeClient, deployment)
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(files).To(HaveExactElements(extensionsv1alpha1.File{
```

Comment:
You could simplify this to:
```suggestion
				Expect(Translate(ctx, fakeClient, deployment)).To(HaveExactElements(extensionsv1alpha1.File{
```